src/workloads/execution/MatchFilters.yml

```diff
@@ -0,0 +1,342 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: >
+  This workload tests a set of match filters.
+
+Keywords:
+  - Loader
+  - CrudActor
+  - QuiesceActor
+  - insert
+  - find
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection Collection0
+  DocumentCount: &DocumentCount 1e6
+  Repeat: &Repeat 10
+  Threads: &Threads 1
+  MaxPhases: &MaxPhases 16
+
+Actors:
+# Phase 0: Insert documents into the collection.
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document:
+          # NOTE: These documents will not produce a match for any of the filters below, forcing
+          # each query to scan the entire collection.
+          a: 0
+          simple_arr: [1,2,3,4,5,6,7,8,9,10]
+          arr_with_object_elements: [{b: 0, c: 0}, {b: 0, c: 0}, {b: 0, c: 0},{b: 0, c: 0},{b: 0,c: 0},{b: 0, c: 0}, {b: 0, c: 0},{b: 0, c: 0},{b: 0, c: 0},{b: 0, c: 0}]
+          str: "this string should never match :)"
+          bit_data: 15
+
+# Phase 1: Ensure all data is synced to disk.
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+
+# Phase 2-N: Run various queries with complex filters.
+- Name: InWithRegexes
+  Type: CrudActor
+  Database: *Database
+  Threads: *Threads
+  Phases:
+   OnlyActiveInPhases:
+     Active: [2]
+     NopInPhasesUpTo: *MaxPhases
+     PhaseConfig:
+       Repeat: *Repeat
+       Collection: *Collection
+       Operations:
+         - OperationName: find
+           OperationCommand:
+             Filter: {"str": {$in: ["a", "b", /^c/, /^d/, /e$/, /f$/]}}
+
+- Name: ElemMatchObject
+  Type: CrudActor
+  Database: *Database
+  Threads: *Threads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [3]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: *Repeat
+        Collection: *Collection
+        Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter: {"arr_with_object_elements": {$elemMatch: {$or: [{b: 1}, {c: 1}, {b: 2}, {c: 2}]}}}
+
+- Name: ElemMatchObjectWide
```

Comment:
Suggest to add `Match` prefix to this workload and the ones above it as well to match the naming with the rest of the file.

Reply:
Done